### PR TITLE
Update to babel/parser 7.15.0

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Description

https://babel.dev/blog/2021/07/26/7.15.0

This PR just do update. I'll support new features at other PRs.

## Checklist

- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-js/parse/babel.js [hunk 1]
```diff
@@ -107,6 +107,13 @@ function createParse(parseMethod, ...optionsCombinations) {
       }));
     }
 
+    if (text.includes("#{") || text.includes("#[")) {
+      combinations = combinations.map((options) => ({
+        ...options,
+        plugins: [...options.plugins, recordAndTuplePlugin],
+      }));
+    }
+
```
Comment (sosukesuzuki) [inline src/language-js/parse/babel.js:116]: Starting with babel 7.15.0, recordAndTuple now conflicts with pipeline.

**Comment by fisker** [comment]
Let's remove `allowAwaitOutsideFunction` in this one.

File: src/language-js/parse/babel.js [hunk 2]
```diff
@@ -107,6 +107,13 @@ function createParse(parseMethod, ...optionsCombinations) {
       }));
     }
 
+    if (text.includes("#{") || text.includes("#[")) {
+      combinations = combinations.map((options) => ({
+        ...options,
+        plugins: [...options.plugins, recordAndTuplePlugin],
+      }));
```
Comment (fisker) [inline src/language-js/parse/babel.js:114]: Let's use `appendPlugins(options, [recordAndTuplePlugin])`, as it's more clear,  and same for the pipeline one below. It should work

Comment (fisker) [reply src/language-js/parse/babel.js:114]: Ah, `appendPlugins` don't accept `options`...

File: tests/format/misc/errors/js/smart-pipeline/__snapshots__/jsfmt.spec.js.snap [hunk 3]
```diff
@@ -55,7 +55,7 @@ exports[`snippet: #8 [babel] format 1`] = `
 `;
 
 exports[`snippet: #9 [babel] format 1`] = `
-"Can not use 'yield' as identifier inside a generator. (1:39)
+"Unexpected yield after pipeline body; any yield expression acting as Hack-style pipe body must be parenthesized due to its loose operator precedence. (1:39)
```
Comment (sosukesuzuki) [inline tests/format/misc/errors/js/smart-pipeline/__snapshots__/jsfmt.spec.js.snap:58]: I don't know well about this. I'll investigate.

Comment (fisker) [reply tests/format/misc/errors/js/smart-pipeline/__snapshots__/jsfmt.spec.js.snap:58]: I think it's fine to update the error message.

File: src/language-js/parse/babel.js (hunk 2 above)
Comment (fisker) [reply src/language-js/parse/babel.js:112]: What do you think, change `appendPlugins` to 

```js
const appendPlugins = (plugins, options = parseOptions) => ({
  ...options ,
  plugins: [...options .plugins, ...plugins],
});
```

then use it?

**Review by fisker — APPROVED** [review]

File: src/language-js/parse/babel.js (hunk 2 above)
Comment (sosukesuzuki) [reply src/language-js/parse/babel.js]: 976be28

**Review by fisker — APPROVED** [review]

**Comment by fisker** [comment]
I wounder if the CI failure caused by https://github.com/prettier/prettier/pull/10859#issuecomment-839882213, seems happens lot recently.

**Pull request merged**